placement/cluster_mapping_provider.go

```diff
@@ -106,21 +96,34 @@ func NewClusterMappingProvider(
 
 // QueryMappings returns the mappings for a given shard and retention policy
 func (mp *clusterMappingProvider) QueryMappings(
-	shard uint32, start, end time.Time) storage.ClusterMappingIter {
-
+	shard uint32, start, end time.Time) (storage.ClusterMappingIter, error) {
```

Comment:
The interface now returns an error so that providers have the option of doing so.  This particular provider does not (yet).